### DISCUSSION TITLE
Render JSON file in MDX file

### DISCUSSION THREAD

**Discussion by supananplus** [mdx-js/mdx#2459] [Q&A]
If I store data of markdown syntax like this:
```json
{
  "type": "root",
  "children": [
    {
      "type": "heading",
      "depth": 1,
      "children": [
        {
          "type": "text",
          "value": "Hello, world!"
        }
      ]
    },
    {
      "type": "blockquote",
      "children": [
        {
          "type": "paragraph",
          "children": [
            {
              "type": "text",
              "value": "This is "
            },
            {
              "type": "strong",
              "children": [
                {
                  "type": "text",
                  "value": "bold"
                }
              ]
            },
            {
              "type": "text",
              "value": " quote"
            }
          ]
        }
      ]
    }
  ]
}
```

When I import to my MDX file it should render as markdown like this

# Hello, world!

> This is **bold** quote 

Can I do that? Or is there another easier way to store and display?

**Comment by ChristianMurphy** [comment]
Welcome @supananplus! 👋 
MDX can already import and include other MDX documents, this is supported in many bundlers and build systems https://mdxjs.com/docs/getting-started/
You don't need to convert it to JSON first, that is an internal representation of MDX that generally isn't put into a file.
Though if you have a very specific and intentional use case for JSON you could use utils to render that back to markdown.

Which comes back to a very important question. What are you actually trying to do?
You've shared your proposed solution without sharing what you are actually trying to solve (an XY Problem https://xyproblem.info/)

**Reply by supananplus** [reply]
Sorry for the inconvenience,
You can assume that I want to make an article website. I receive data from my server and it comes in JSON format I have to figure out what structure of my JSON will look like and how to render it as markdown so that's why. 

Is it clearer for you? English isn't my mother language but I try to make it clear for you as much as I can.

**Reply by ChristianMurphy** [reply]
> You can assume that I want to make an article website. I receive data from my server and it comes in JSON format I have to figure out what structure of my JSON will look like and how to render it as markdown so that's why.

If it's your server you could store it as anything you want right?
Why do you specifically want it in JSON?

---

To answer your direct question.
It is possible to use `unified`, `remark-rehype`, and `rehype-react` to go from MDAST JSON to react components you can use on the page.
https://unifiedjs.com/learn/
